lib/value-pairs/value-pairs.c

```diff
@@ -601,34 +601,42 @@ vp_walker_skip_sdata_enterprise_id(const gchar *name, gint pos)
   do
     {
       /* skip @ or . */
-      pos++;
-      pos += strspn(&name[pos], "0123456789");
+      ++name;
+      name += strspn(name, "0123456789");
     }
-  while (name[pos] == '.' && isdigit(name[pos + 1]));
-  return pos;
+  while (*name == '.' && isdigit(*(name + 1)));
+  return name;
 }
 
 static GPtrArray *
 vp_walker_split_name_to_tokens(vp_walk_state_t *state, const gchar *name)
 {
-  gint pos, token_start = 0, name_len = strlen(name);
+  const gchar *token_start = name;
+  const gchar *token_end = name;
+
   GPtrArray *array = g_ptr_array_sized_new(VP_STACK_INITIAL_SIZE);
 
-  pos = 0;
-  do
+  while (*token_end)
     {
-      token_start = pos;
-      pos += strcspn(&name[pos], "@.");
-      if (name[pos] == '@')
-        pos = vp_walker_skip_sdata_enterprise_id(name, pos);
-
-      if (name[pos] == '.' || pos == name_len)
+      if (*token_end == '@')
```

Comment:
The body of the loop can be rewritten to a simple switch-case statement. Should it be done?